pkg/cache/context.go

```diff
@@ -702,10 +702,19 @@ func (ctx *Context) updateApplicationTags(request *interfaces.AddApplicationRequ
 	// add resource quota info as an app tag
 	resourceQuota := utils.GetNamespaceQuotaFromAnnotation(namespaceObj)
 	if resourceQuota != nil && !common.IsZero(resourceQuota) {
-		if quotaStr, err := json.Marshal(resourceQuota); err == nil {
-			request.Metadata.Tags[siCommon.AppTagNamespaceResourceQuota] = string(quotaStr)
+		if quota, err := json.Marshal(resourceQuota); err == nil {
+			request.Metadata.Tags[siCommon.AppTagNamespaceResourceQuota] = string(quota)
 		}
 	}
+
+	// add guaranteed resource info as an app tag
+	guaranteedResource := utils.GetNamespaceGuaranteedFromAnnotation(namespaceObj)
+	if guaranteedResource != nil && !common.IsZero(guaranteedResource) {
+		if guaranteed, err := json.Marshal(guaranteedResource); err == nil {
+			request.Metadata.Tags[siCommon.AppTagNamespaceResourceGuaranteed] = string(guaranteed)
+		}
+	}
```

Comment:
Please add test coverage to `TestAddApplicationsWithTags()`, there is a dummy lister and existing namespace object which you can re-use.